go/wavefrontsender.md

```diff
@@ -0,0 +1,186 @@
+
+| `ProxyConfiguration` field | Corresponding property in `wavefront.conf` |
+| ----- | -------- |
+| `MetricsPort` | `pushListenerPorts=` |
+| `DistributionPort` | `histogramDistListenerPorts=` |
+| `TracingPort` | `traceListenerPorts=` |
+ 
+
+## Option 2: Sending Data via Direct ingestion
+
+When sending data via direct ingestion, you need to create a direct `Sender` and configure it to send data directly to Wavefront.
+
+>**Prerequisite**
+> * Verify that you have the Direct Data Ingestion permission. For details, see [Examine Groups and Permissions](https://docs.wavefront.com/users_account_managing.html#examine-groups-and-permissions).
+> * The URL of your Wavefront instance. This is the URL you connect to when you log in to Wavefront, typically something like `https://<domain>.wavefront.com`.
+> * [Obtain the API token](http://docs.wavefront.com/wavefront_api.html#generating-an-api-token).
+
+### Configure the Direct Sender
+Create a `DirectConfiguration` that has the server and token information you obtained. 
+
+You can optionally tune the following ingestion properties:
+
+* Max buffer size - Internal buffer capacity of the direct `Sender`, which helps handle brief increases in data and buffering on errors. Any data in excess of this size is dropped.  Separate buffers are maintained per data type (metrics, spans, and distributions).
+* Flush interval - Interval for flushing data from the direct `Sender` directly to Wavefront.
+* Batch size - Amount of data to send to Wavefront in each flush interval.
+
+Together, the flush interval and batch size control the maximum theoretical throughput of the direct `Sender`. Override the defaults _only_ to set higher values.
+
+```go
+import (
+    time
+    "github.com/wavefronthq/wavefront-sdk-go/senders"
+)
+
+func main() {
+    directCfg := &senders.DirectConfiguration {
+        // Your Wavefront instance URL
+        Server : "https://INSTANCE.wavefront.com", 
+        
+        // Wavefront API token created with direct ingestion permission
+        Token : "YOUR_API_TOKEN",
+
+        // Optional: Override the batch size (in data points). Default: 10,000. Recommended not to exceed 40,000.
+        BatchSize : 20000,
+
+        // Optional: Override the max buffer size (in data points). Default: 50,000. Higher values could use more memory.
+        MaxBufferSize : 50000,
+
+        // Optional: Override the flush interval (in seconds). Default: 1 second
+        FlushIntervalSeconds : 2,
+    }
+
+    // Create the direct sender
+    sender, err := senders.NewDirectSender(directCfg)
+    if err != nil {
+        // handle error
+    }
+    ... // Use the direct sender to send data 
+}
+```
+
+
+## Share a Wavefront Sender
+
+Various Wavefront SDKs for Go use this library and require a `Sender` instance.
+
+If you are using multiple Wavefront Go SDKs within the same process, you can create a single `Sender` and share it among the SDKs. 
+ 
+<!--- 
```

Comment:
@sushantdewan123 : Can we get an example similar to the Java content, here: https://github.com/wavefrontHQ/wavefront-sdk-doc-sources/blob/master/java/wavefrontsender.md#share-a-wavefrontsender
What do you think?